### PR TITLE
`Prg` benchmarks

### PR DESCRIPTION
**Pull request by cjpatton**
Benchmarking different methods for generating psuedorandom sequences of field elements in libprio.

### CODE REVIEW

File: src/field.rs [hunk 1]
```diff
@@ -624,6 +650,20 @@ pub fn random_vector<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
...
 }
 
+/// Generate a vector of uniform random field elements.
+pub fn random_vector2<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
+    Ok(Prng2::generate(Suite::Aes128CtrHmacSha256)?
+        .take(len)
+        .collect())
+}
+
+/// Generate a vector of uniform random field elements.
```
Comment (cjpatton) [inline src/field.rs]: This is https://github.com/cjpatton/vdaf/pull/29, right? Can you rename the function to something more meaningful, such as "random_vector_pad_then_reduce" or something?

File: src/field.rs [hunk 2]
```diff
@@ -624,6 +650,20 @@ pub fn random_vector<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
         .collect())
 }
 
+/// Generate a vector of uniform random field elements.
```
Comment (cjpatton) [inline src/field.rs]: This is https://github.com/cjpatton/vdaf/pull/30, right? Can you give this a more meaningful name, to help me keep this straight?

File: src/prng.rs [hunk 3]
```diff
@@ -20,6 +20,140 @@ pub enum PrngError {
...
+            output_written: 0,
+        }
+    }
+
+    pub(crate) fn get(&mut self) -> F {
+        loop {
+            // Seek to the next chunk of the buffer that encodes an element of F.
+            for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
+                let j = i + F::ENCODED_SIZE;
+                if let Some(x) = match F::hash_to_field(&self.buffer[i..j]) {
```
Comment (cjpatton) [inline src/prng.rs]: It looks like what we're doing here is generating a pseudorandom string from AES-CTR, then applying hash_to_field directly to the output. This isn't what I intended for us to compare. The fact that this isn't initially clear is my fault. I'm sorry about that. Let me try to be more clear.

What we want to measure is the performance of the algorithm specified [here](https://github.com/cjpatton/vdaf/pull/29), which transforms a byte string into a sequence of field elements. This algorithm does not apply hash_to_field directly, but rather a variant that replaces `expand_message(msg, DST, len_in_bytes)` (see https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-13.html#name-hash_to_field-implementatio) with the algorithm currently used in libprio for expanding a seed into a byte string of the requested length.

In other words, `random_vector3` should look something like this:
```rust
pub fn random_vector3<F: FieldElement>(len: usize, padding: usize) -> Result<Vec<F>, PrngError> {
    let key = Key::generate(Suite::Aes128CtrHmacSha256)?;
    let key_stream = KeyStream::from_key(&key);
    let mut buf = vec![0; len * (F::ENCODED_SIZE + padding)];
    key_stream.fill(&mut buf);
    Ok(buf.chunks(F::ENCODED_SIZE + padding).map(|chunk| F::from_be_bytes(chunk).unwrap()).collect())
}
```

(By the way, there's no need to re-implement the `Prng` API. We'll probably want something like this later, but I suggest we keep it simple for now.)

File: src/prng.rs [hunk 4]
```diff
@@ -20,6 +20,140 @@ pub enum PrngError {
...
+            output_written: 0,
+        }
+    }
+
+    pub(crate) fn get(&mut self) -> F {
+        loop {
+            // Seek to the next chunk of the buffer that encodes an element of F.
+            for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
+                let j = i + F::ENCODED_SIZE;
+                if let Some(x) = match F::from_be_bytes(&self.buffer[i..j]) {
```
Comment (cjpatton) [inline src/prng.rs]: It looks like this is something other than https://github.com/cjpatton/vdaf/pull/29 or https://github.com/cjpatton/vdaf/pull/30.

Comment (armfazh) [reply src/prng.rs]: indeed, changed now

File: src/prng.rs (hunk 3 above)
Comment (armfazh) [reply src/prng.rs]: your snippet is much simpler, now is refactored.
